networking/hub-default.json

```diff
@@ -84,84 +302,121 @@
-                                    "service": "Microsoft.KeyVault"
-                                }
-                            ]
+                            "addressPrefix": "[parameters('azureFirewallSubnetAddressSpace')]"
                         }
                     },
                     {
                         "name": "GatewaySubnet",
                         "properties": {
-                            "addressPrefix": "[parameters('azureGatewaySubnetAddressSpace')]",
-                            "serviceEndpoints": [
-                            ]
+                            "addressPrefix": "[parameters('azureGatewaySubnetAddressSpace')]"
                         }
                     },
                     {
                         "name": "AzureBastionSubnet",
                         "properties": {
                             "addressPrefix": "[parameters('azureBastionSubnetAddressSpace')]",
-                            "serviceEndpoints": [
-                            ]
+                            "networkSecurityGroup": {
+                                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('bastionNetworkNsgName'))]"
+                            }
                         }
                     }
-                ],
-                "enableDdosProtection": false,
-                "enableVmProtection": false
-            }
+                ]
+            },
+            "resources": [
+                {
+                    "type": "providers/diagnosticSettings",
+                    "apiVersion": "2017-05-01-preview",
+                    "name": "Microsoft.Insights/default",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.OperationalInsights/workspaces', variables('hubLaName'))]",
+                        "[resourceId('Microsoft.Network/virtualNetworks', variables('hubVnetName'))]"
+                    ],
+                    "properties": {
+                        "workspaceId": "[resourceId('Microsoft.OperationalInsights/workspaces', variables('hubLaName'))]",
+                        "metrics": [
+                            {
+                                "category": "AllMetrics",
+                                "enabled": true
+                            }
+                        ]
+                    }
+                }
+            ]
         },
         {
-            "apiVersion": "2019-11-01",
             "type": "Microsoft.Network/publicIpAddresses",
-            "name": "[variables('defaultFwPipName')]",
+            "apiVersion": "2020-05-01",
+            "name": "[variables('hubFwPipNames')[copyIndex()]]",
             "location": "[parameters('location')]",
+            "comments": "This is a public IP for this regional hub's firewall. You'll want as many as necessary to avoid SNAT port exhaustion. Typical production hubs may require 20 IPs -- we only deploy three here as a reference.",
```

Comment:
It would be good to include a link to give more info on SNAT, not sure if https://docs.microsoft.com/azure/load-balancer/load-balancer-outbound-connections#scenarios would be good, or if there's one more specific to the scenario

Reply:
We don't have good documentation around this that would be reasonable to link here for Azure Firewall.  All references (that I'm aware of) to it are "in passing" unfortunately.